EmbodiedSuffering_Engine/Compute/SufferingIndex.cs

```diff
@@ -0,0 +1,115 @@
+        [MultiOutput(2, "invalidCountries", "Countries that had no data for NumberOfEnslavedPeople.")]
+        public static Output<double, List<string>, List<string>> SufferingIndex(List<double> numberOfEnslavedPeople, MaterialImportSources materialImportSources, double acceptableThreshold)
+        {
+            double sufferingIndex = 0;
+            List<double> importRatios = materialImportSources.ImportRatios.ToList();
+            List<List<Country>> exportCountries = new List<List<Country>>();
+            List<string> culledCountries = new List<string>();
+            List<string> invalidCountries = new List<string>();
+
+            // List length check: Fail the calculation if the lists are of unequal lengths
+            if (numberOfEnslavedPeople.Count != importRatios.Count)
+            {
+                int numCount = numberOfEnslavedPeople.Count;
+                int ratiosCount = importRatios.Count;
+                BH.Engine.Base.Compute.RecordError("The provided list's lengths to do correspond as expected." + $"\nNumberOfEnslavedPeople count = {numCount}" + $"\nImportRatios count = {ratiosCount}. \nPlease update your data and try again.");
+            } else
+            {
+                // Remove after debugging is complete
+                BH.Engine.Base.Compute.RecordNote("Lists are the same length, proceeding to threshold check.");
+            }
+
+            // Add countries to ImportCountries[]
+            for (int i = 0; i < materialImportSources.ImportRatios.Count; i++)
+            {
+                exportCountries.Add(materialImportSources.ExportCountries);
+            }
+
+            // Pass a list of string import countries
+            // This is used in the threshold check
+            List<string> importCountriesAsStrings = exportCountries.ConvertAll(x => x.ToString());
+
+            // Threshold check: Toss anything above the input threshold into badCountries[]
+            for (int i = 0; i < numberOfEnslavedPeople.Count; i++)
+            {
+                if (numberOfEnslavedPeople[i] > acceptableThreshold)
+                {
+                    culledCountries.Add(importCountriesAsStrings[i]);
+                    numberOfEnslavedPeople.Remove(numberOfEnslavedPeople[i]);
+                    BH.Engine.Base.Compute.RecordWarning("Some countries were removed from your calculation due to exceeding the acceptableThreshold requirement. \nPlease review them from the CulledCountries output. \nThe calculation will now proceed with all remaining countries.");
+                }
+            }
+
+            // Data validitiy check: Toss anything with 0 or NaN into invalidCountries[]
+            for (int i = 0; i < numberOfEnslavedPeople.Count; i++)
+            {
+                if (numberOfEnslavedPeople[i] == double.NaN)
+                {
+                    numberOfEnslavedPeople[i] = 0;
```

Comment:
From my tests, this has no effect. I could try `double.IsNaN()`, but from what I've seen there could be values just missing entirely from the dataset. This would require lots more null handling types.